### ISSUE TITLE
conda env update leads to broken environments

### ISSUE THREAD

**Issue by Korijn** [conda/conda#3650]
A very simple reproduction:

Create the following `environment.yml` file in a directory of your choice:

```
name: condaenvbug
dependencies:
  - python
  - vtk =5.10.1
```

> conda env create
> activate condaenvbug

Change `environment.yml` by bumping VTK to a different version:

```
name: condaenvbug
dependencies:
  - python
  - vtk =6.3.0
```

> (condaenvbug) conda env update
> (condaenvbug) conda list

```
(condaenvbug) C:\Users\Korijn\dev\condabug>conda list
# packages in environment at C:\Users\Korijn\Miniconda3\envs\condaenvbug:
#
pip                       8.1.2                    py27_0    defaults
python                    2.7.12                        0    defaults
setuptools                27.2.0                   py27_1    defaults
vs2008_runtime            9.00.30729.1                  2    defaults
vtk                       5.10.1                   py27_0    defaults
vtk                       6.3.0                    py27_1    defaults
wheel                     0.29.0                   py27_0    defaults
```

As you can see there are now two installed VTK packages. In our bigger projects this simply destroys the environment and makes it unusable. We have fallen back to always calling `conda env remove -n {myenv} -y && conda env create` for now.

_This is not specific to the VTK package, it was just the first one that came to mind for this example._

---

> conda info

```
C:\Users\Korijn\dev\condabug>conda info
Current conda install:

               platform : win-64
          conda version : 4.2.9
       conda is private : False
      conda-env version : 4.2.9
    conda-build version : 2.0.6
         python version : 3.5.1.final.0
       requests version : 2.9.1
       root environment : C:\Users\Korijn\Miniconda3  (writable)
    default environment : C:\Users\Korijn\Miniconda3
       envs directories : C:\Users\Korijn\Miniconda3\envs
          package cache : C:\Users\Korijn\Miniconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/win-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
                          https://repo.continuum.io/pkgs/msys2/win-64/
                          https://repo.continuum.io/pkgs/msys2/noarch/
            config file : C:\Users\Korijn\.condarc
           offline mode : False
```

**Comment by ma-ba** [comment]
Should be a duplicate of #3478.

**Comment by Korijn** [comment]
Definitely looks like it.

**Comment by kalefranz** [comment]
I'm hoping https://github.com/conda/conda/pull/3704 will fix this issue.  But if someone wants to turn @Korijn's description into a conda-env integration test case, I'd be happy to merge the PR!

**Comment by Korijn** [comment]
Do you have an example of such a test that I can copy & adjust?

**Comment by kalefranz** [comment]
There are some [here](https://github.com/conda/conda/blob/master/tests/conda_env/test_create.py) and [here](https://github.com/conda/conda/blob/master/tests/conda_env/test_cli.py).  Modeled after the integration tests [here](https://github.com/conda/conda/blob/master/tests/test_create.py).

**Comment by Korijn** [comment]
This appears to be fixed with conda 4.2.12! I'll submit a regression test PR shortly.

**Comment by Korijn** [comment]
I wanted to add some integration tests, but there already is one that perfectly describes the case here:

https://github.com/conda/conda/commit/b839123e6c3a38d45b72711c89c688ce7ec71ba3#diff-f97723fb32d92c38a3488c80037d07d0R103

So I'm closing this. Nice work everyone!
